### PR TITLE
Make writeStackCookie() use HEAPU32

### LINKED ISSUE #16176: HEAPU32 used instead of HEAP32 in checkStackCookie
I guess this is not too important, but in a runtime generated with:
```
em++ -pthread -fwasm-exceptions -s PTHREAD_POOL_SIZE=4 -s MINIMAL_RUNTIME=1 --minify 0 code.cpp -o code.html
```
`writeStackCookie` is using `HEAP32`, while `checkStackCookie` is using `HEAPU32`. 

```
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.0.1 (5ebe1d421445fa9a4a02c1efc6d42bde3ac90da2)
clang version 14.0.0 (https://github.com/llvm/llvm-project d98dfb2baabb647a6ac4c4f4399999cfc44433db)
Target: wasm32-unknown-emscripten
Thread model: posix
```

### PR DESCRIPTION
**Pull request by benjitrosch**
Following up on comments left by @sbc100  on #16433 to close #16176

### CODE REVIEW

File: src/runtime_stack_check.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ function writeStackCookie() {
   // The stack grow downwards towards _emscripten_stack_get_end.
   // We write cookies to the final two words in the stack and detect if they are
   // ever overwritten.
-  {{{ makeSetValue('max', 0, '0x2135467', 'i32' ) }}};
-  {{{ makeSetValue('max', 4, '0x89BACDFE', 'i32' ) }}};
```
Comment (sbc100) [inline src/runtime_stack_check.js:21]: I think you can change these from `i32` to `u32` to get the result you want.

Comment (benjitrosch) [reply src/runtime_stack_check.js:21]: @sbc100 Done, thanks!

**Comment by benjitrosch** [comment]
Seems to be failing only for the Firefox test with

```AssertionError: Expected to find '/report_result?exit:0
' in '[no http server activity]
', diff:```

Any idea what might be causing this?

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
(yes, looks like the firefox error was a random flake on CI)

**Pull request merged**